feat: conclude set-icon experiment (#623)
slackapi/slack-cli: docs/reference/experiments.md

- `lipgloss`: shows pretty styles.
- `manifest-sync`: resolves conflicting app manifest values.

## Experiments changelog

Below is a list of updates related to experiments.

> - **August 2026**: Concluded the `set-icon` experiment with full support for icon upload on all app types now enabled by default in the Slack CLI.

Reviewer: ```suggestion
- **August 2026**: Concluded the `set-icon` experiment; The Slack CLI now offers full support for icon upload on all app types by default.
```

just a little easier to read methinks
